src/vconsole/vconsole-setup.c

```diff
@@ -537,28 +544,33 @@ static int find_source_vc(char **ret_path, unsigned *ret_idx) {
 
                 fd = open_terminal(path, O_RDWR|O_CLOEXEC|O_NOCTTY);
                 if (fd < 0) {
-                        RET_GATHER(err, log_debug_errno(fd, "Failed to open terminal %s, ignoring: %m", path));
+                        log_debug_errno(fd, "Failed to open terminal %s, ignoring: %m", path);
+                        if (IN_SET(err, 0, -EBUSY, -ENOENT))
+                                err = fd;
                         continue;
                 }
 
                 r = verify_vc_kbmode(fd);
+                if (r >= 0)
+                        r = verify_vc_display_mode(fd);
                 if (r < 0) {
-                        RET_GATHER(err, log_debug_errno(r, "Failed to check VC %s keyboard mode: %m", path));
+                        log_debug_errno(r, "Failed to check VC %s keyboard or display mode: %m", path);
+                        if (IN_SET(err, 0, -ENOENT))
```

Comment:
why nest this check? seems nice as it was before?

Reply:
simply to share the handling of the error case.

Reply:
yeah, but the log message is less accurate then? i prefer the status quo ante...